iModelCore/ecobjects/src/ECProperty.cpp

```diff
@@ -1399,7 +1399,7 @@ ECObjectsStatus PrimitiveECProperty::SetType (ECEnumerationCR enumerationType)
     {
     if (&(enumerationType.GetSchema()) != &(this->GetClass().GetSchema()))
         {
-        if (!ECSchema::IsSchemaReferenced(this->GetClass().GetSchema(), enumerationType.GetSchema()))
+        if (!ECSchema::IsSchemaReferenced(this->GetClass().GetSchema(), enumerationType.GetSchema(), ECN::SchemaMatchType::Exact, true))

```

Comment:
A schema must directly reference a schema to use something from it.  The schema that owns the enumeration must be added as a reference before the type can be set.